File: scripts/update-latest-yml.py
#!/usr/bin/env python3
"""Update latest.yml and latest-linux.yml to v1.54.0."""
import os

# Windows latest.yml content
WINDOWS_YML = """version: 1.54.0
releaseDate: 2026-05-24T10:00:00.000Z
files:
  - url: LingJing-Setup-1.54.0-win-x64.exe
    sha512: 202900ead30f338febaa56bcbcefafae383fa789615bfa1eac4416d419a998422011f7ed08184ba2b49f5b04415e5a8117f6ed95802ef57888d85fce55899a78
    size: 142143591
path: LingJing-Setup-1.54.0-win-x64.exe
sha512: 202900ead30f338febaa56bcbcefafae383fa789615bfa1eac4416d419a998422011f7ed08184ba2b49f5b04415e5a8117f6ed95802ef57888d85fce55899a78
releaseNotes: v1.54.0: 全面系统检查修复 - electron-builder清理 + 全平台版本升级部署
"""

# Linux latest-linux.yml content
LINUX_YML = """version: 1.54.0
releaseDate: 2026-05-24T10:00:00.000Z
files:
  - url: LingJing-1.54.0-linux-x86_64.AppImage
    sha512: 23ff36f671cc192f9eab4e07a7cdc20abbbc19b9cb5101d0594c7564927514f1699f4986d1626ebc0f8f479c31a3f524a5978d136edcd52d69f970bef891ac6a
    size: 180006466
path: LingJing-1.54.0-linux-x86_64.AppImage
sha512: 23ff36f671cc192f9eab4e07a7cdc20abbbc19b9cb5101d0594c7564927514f1699f4986d1626ebc0f8f479c31a3f524a5978d136edcd52d69f970bef891ac6a
releaseNotes: v1.54.0: 全面系统检查修复 - electron-builder清理 + 全平台版本升级部署
"""

# Linux locations for latest-linux.yml
LINUX_YML_PATHS = [
    '/var/www/html/downloads/latest-linux.yml',
    '/var/www/html/latest-linux.yml',
    '/var/www/lingjing/latest-linux.yml',
]

# Windows locations for latest.yml
WINDOWS_YML_PATHS = [
    '/var/www/html/downloads/latest.yml',
    '/var/www/html/latest.yml',
    '/var/www/lingjing/latest.yml',
]

def main():
    for path in LINUX_YML_PATHS:
        with open(path, 'w') as f:
            f.write(LINUX_YML)
        print(f"  ✅ {path}")

    for path in WINDOWS_YML_PATHS:
        with open(path, 'w') as f:
            f.write(WINDOWS_YML)
        print(f"  ✅ {path}")

    print("\n✅ All YML files updated to v1.54.0")

if __name__ == '__main__':
    main()

File: infra/scripts/update-latest-yml.js
const fs = require('fs');

const winYml = {
  version: '1.55.1',
  files: [
    {
      url: 'LingJing-Setup-1.55.1-win-x64.exe',
      sha512: '3d80d6a8f297730456f872446d8f990afe5172df482906e1982c483a31c8e15d4326aec9d0ec568471f3bb46a51f79d965971dc932000586da01de60dedadff4',
      size: 142145134
    }
  ],
  path: 'LingJing-Setup-1.55.1-win-x64.exe',
  sha512: '3d80d6a8f297730456f872446d8f990afe5172df482906e1982c483a31c8e15d4326aec9d0ec568471f3bb46a51f79d965971dc932000586da01de60dedadff4',
  releaseDate: new Date().toISOString()
};

const linYml = {
  version: '1.55.1',
  files: [
    {
      url: 'LingJing-1.55.0-linux-x86_64.AppImage',
      sha512: '79767a1ac8dd691c010a765517892c7594b5bf8006f8fe65bc71f39656df43978eacfc91e141e58ccb660dfa5cce0cea73159c20379eb2c806ef1f6f5246a3f0',
      size: 182918602
    }
  ],
  path: 'LingJing-1.55.0-linux-x86_64.AppImage',
  sha512: '79767a1ac8dd691c010a765517892c7594b5bf8006f8fe65bc71f39656df43978eacfc91e141e58ccb660dfa5cce0cea73159c20379eb2c806ef1f6f5246a3f0',
  releaseDate: new Date().toISOString()
};

function toYaml(obj, indent) {
  indent = indent || '';
  let result = '';
  for (const [key, value] of Object.entries(obj)) {
    if (value === null || value === undefined) continue;
    if (Array.isArray(value)) {
      result += indent + key + ':\n';
      for (const item of value) {
        if (typeof item === 'object') {
          result += indent + '  - ' + Object.entries(item).map(([k, v]) => k + ': ' + JSON.stringify(v)).join('\n' + indent + '    ') + '\n';
        }
      }
    } else if (typeof value === 'object') {
      result += indent + key + ':\n';
      result += toYaml(value, indent + '  ');
    } else {
      result += indent + key + ': ' + JSON.stringify(value) + '\n';
    }
  }
  return result;
}

const paths = ['/var/www/downloads/latest.yml', '/var/www/html/latest.yml', '/var/www/lingjing/latest.yml'];
for (const p of paths) {
  fs.writeFileSync(p, toYaml(winYml));
  console.log('UPDATED:', p);
}

const linPaths = ['/var/www/downloads/latest-linux.yml', '/var/www/html/latest-linux.yml', '/var/www/lingjing/latest-linux.yml'];
for (const p of linPaths) {
  fs.writeFileSync(p, toYaml(linYml));
  console.log('UPDATED:', p);
}
